### PR TITLE
add ambient/bounding set fields to user records, and set CAP_WAKE_ALARM by default for local regular user sessions

### PR DESCRIPTION
**Pull request by poettering**
Fixes: #17564 #21382

### CODE REVIEW

File: mkosi.conf.d/fedora/10-fedora.conf [hunk 1]
```diff
@@ -24,6 +24,7 @@ Packages=
         libasan
         libbpf
         libcap-ng
+        libcap-ng-utils
```
Comment (daandemeyer) [inline mkosi.conf.d/fedora/10-fedora.conf:27]: I went through a lot of effort to get the packages in sync across all distros so can we please add these for the other distros as well. https://pkgs.org/ is quite nice for searching packages

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

File: src/shared/user-record-show.c [hunk 2]
```diff
@@ -287,6 +288,22 @@ void user_record_show(UserRecord *hr, bool show_full_group_info) {
         if (hr->access_mode != MODE_INVALID)
                 printf(" Access Mode: 0%03o\n", user_record_access_mode(hr));
 
+        uint64_t caps = user_record_capability_bounding_set(hr);
+        if (caps != UINT64_MAX) {
+                _cleanup_free_ char *scaps = NULL;
+
+                (void) capability_set_to_string_negative(caps, &scaps);
+                printf(" Bound. Caps: %s\n", scaps);
```
Comment (dnicolodi) [inline src/shared/user-record-show.c]: If "Bound." is an abbreviation, I don't understand what it abbreviates.

Comment (poettering) [reply src/shared/user-record-show.c]: "The capability bounding set", as per capabilities(7)

Comment (bluca) [inline src/shared/user-record-show.c]: `scaps` might be NULL here, so either strna(scaps) or check for errors?

File: man/homectl.xml [hunk 3]
```diff
@@ -271,6 +271,19 @@
         the user will be removed from the group.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--capability-bounding-set=</option><replaceable>CAPABILITIES</replaceable></term>
+        <term><option>--capability-ambient-set=</option><replaceable>CAPABILITIES</replaceable></term>
+
+        <listitem><para>These options that a space separated list of process capabilities
```
Comment (bluca) [inline man/homectl.xml]: /that/take/

File: src/login/pam_systemd.c [hunk 4]
```diff
@@ -47,20 +49,79 @@
...
+                pam_handle_t *handle,
+                const char *value,
+                uint64_t *caps) {
+
+        bool subtract;
+        int r;
+
+        assert(handle);
+        assert(value);
+
```
Comment (bluca) [inline src/login/pam_systemd.c:62]: if `caps` is NULL this function will never do anything, no? just short-cut it here then?

**Review by bluca — APPROVED** [review]

File: src/login/pam_systemd.c (hunk 4 above)
Comment (poettering) [reply src/login/pam_systemd.c:62]: it will validate them, and generate log messages.

**Comment by sc0w** [comment]
NOTE: https://github.com/systemd/systemd/issues/21382 wasn't closed after the merge, seems it needs to close manually

**Pull request merged**